src/main/java/frc/robot/commands/BangBangDriveCommand.java

```diff
@@ -0,0 +1,72 @@
+package frc.robot.commands;
+
+import edu.wpi.first.math.Pair;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.constants.DriveConstants;
+import frc.robot.subsystems.Drivetrain;
+import java.awt.geom.Point2D;
+
+public class BangBangDriveCommand extends Command{
+    Drivetrain driv;
+    Pose2d setpoint;
+    int tim;
+    public BangBangDriveCommand(Drivetrain drev, Pose2d setpoint){
+        driv = drev;
+        this.setpoint = setpoint;
+    }
+
+    @Override
+    public void initialize(){
+        driv.resetEncoders();
+        tim = 0;
+    }
+
+    @Override
+    public void execute(){
+        tim ++;
+        if (tim <= 250){
+            driv.arcadeDrive(0, normalize(calculateTurnAngle(setpoint, (driv.getGyroAngle().getDegrees())), -180, 180));
+        }else{
+            driv.arcadeDrive(1, normalize(calculateTurnAngle(setpoint, (driv.getGyroAngle().getDegrees())), -180, 180));
+
```

Comment:
Don't set throttle to full power

Reply:
Also, thought you were supposed to use getAveragePosition in execute? Kyle mentioned you were making something more advanced with the bang bang command similar to GoToPose but lmk

Reply:
Also, thought you were supposed to use getAveragePosition in execute? It would continuously check the robot's current position relative to its target, allowing for more responsive adjustments, and would be more functional by instead of waiting for a fixed amount of cycles, checking if it's closer to the target

Reply:
Setting throttle to full power is fine because arcadeDrive() calls tankDrive(), which multiplies the inputs by 0.25, so it isn't actually full power on a real robot.

Also, although the instructions say to use getAverageAngle(), this command is slightly more complicated and different form the command described in the instructions. You need to use Bang-Bang control in at least part of the command, which should probably be turning for this command. You can compare getGyroAngle() to the expected angle. Instead of subtracting them and limiting the values, you should use a simple if statement to determine direction. If the values are close (like ±1 degree), then go tot he next step int he command, instead of using a counter.